src/microsim/transportables/MSPModel_JuPedSim.cpp

```diff
@@ -696,12 +723,13 @@ SUMOTime MSPModel_JuPedSim::PState::getWaitingTime(const MSStageMoving& /* stage
 
 
 double MSPModel_JuPedSim::PState::getSpeed(const MSStageMoving& /* stage */) const {
-    return 0;
+    Position velocity = (myPosition - myPreviousPosition) / STEPS2TIME(DELTA_T);
+    return velocity.length2D();
 }
```

Comment:
This is quite inefficient. You could just use myPosition.distanceTo2D(myPreviousPosition) / TS